### DISCUSSION TITLE
Albums support 🎉

### DISCUSSION THREAD

**Discussion by TheLastGimbus** [TheLastGimbus/GooglePhotosTakeoutHelper#187] [Announcements]
# WELL GUYS, WE DID IT 🥳

After 3 years, I finally did it 😌 - it wasn't easy at all, as i've spent my **whole weekend** on this, but here they are - as good as they can be

https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/pull/186

## Solutions
Discussion in #135 wasn't very live, but i came up with **4** solutions myself:
- `shortcut`: Album folders with shortcuts/symlinks to original photos. Recommended as it will take the least space, but may not be portable when moving across systems/computes/phones etc
  
  Yes, properly working shortcuts on windoza! It's super slow because i need to launch powershell every time, but works! This is the part That i'm most worried about and need testing

  On linux/macos it will be symlinks - not really universal to move to other pc, but if you want portability, just use other option

- `nothing`: Just ignore them and put year-photos into one folder

  Note that this will ignore Archive and trash folders, and will just look at year folders

- `duplicate-copy`: Album folders with photos copied into them. This will work across all systems, but may take wayyy more space!!

  I think that's what most of you wanted

- `json`: Put ALL photos (including Archive and Trash) in one folder and make a .json file with info about albums. Use if you're a programmer, or just want to get everything, ignoring lack of year-folders etc.

  this is nice alternative to `nothing` option - you just export **everything** to one folder - including archive

## Testing 🔧

I tested it with my takeouts and wrote some unit tests, but i wasn't born yesterday and i know that those takeouts can get really creative. Thus, i would need you guys to help me test it before i release it officially

Here are `.exe`s with latest `v3.4.0rc1` album-ed version: https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/actions/runs/4461724986 - just go down to Artifacts and chose your OS

If you could check if nothing crashes and it doesn't miss any photos, it would be awesome 💖

@wjhladik @tribulete @omlet05 @LPFchan @shlomis1 @rebelga @michaelkebe @dburner @brock @palijn @isabelpothos

**Comment by tribulete** [comment]
Thanks for yours efforts. I´ll try it and send you the results.

**Comment by TheLastGimbus** [comment]
No bug reports so far so here it is: https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/releases/tag/v3.4.0

**Comment by TheKitBane** [comment]
Hi there, I only want to sort photos in one album from my google photos (I'm doing a video of a specific trip and don't need every single photo in my library). I keep getting the failure message about my google takeout not having year folders and therefore the process failing. Any way around this? it's two weeks of footage, so there wouldn't even be more than one month folder tbh

**Reply by TheLastGimbus** [reply]
Nope, gpth doesn't and won't do "just the one" - the whole process is complicated that it requires year foledrs to, sry :/
